content/nginx-one/nginx-configs/config-templates/import-templates.md

```diff
@@ -0,0 +1,375 @@
+}
+EOF
+# No schema.yaml needed - template has no variables
+
+tar -czf ../health-check.tar.gz *
+cd ..
+
+# Final structure
+ls -la *.tar.gz
+```
+
+### Tips
+
+1. **Do not create nested directories in archives**
+   ```bash
+   # Avoid - nested directory structure is not supported
+   tar -czf template.tar.gz template-dir/
+   
+   # Correct - keep files at root level
+   cd template-dir && tar -czf ../template.tar.gz *
+   ```
+
+2. **Do not include unnecessary files**
+   ```bash
+   # Avoid - hidden files, backups or other file extensions will be ignored
+   tar -czf template.tar.gz *~ .* *.bak *.tmpl schema.yaml
+   
+   # Correct - only include required files
+   tar -czf template.tar.gz *.tmpl schema.yaml
+   ```
+
+3. **Include schema for templates with variables**
+    - If your `.tmpl` file contains `{{ .variable }}`, you must include a schema file
+    - Import will be rejected without proper schema
+
+4. **Do not include multiple template files in one archive**
+    - Each archive must contain exactly one `.tmpl` file
+    - Create separate archives for related templates
+
+### Ready to import
+
+Once you have created your archives, you can import them using the [Import a template API]({{< ref "/nginx-one/api/api-reference-guide/#operation/importTemplate" >}}) operation.
+
+#### Required API parameters
+
+In addition to your template archive, the import API requires several parameters. Read the API specification for full details, but the key parameters include:
+
+**name** (required):
+- A unique name for the template within your organization. Use the same as the archive name for clarity without the extension (e.g., `reverse-proxy-base`, `cors-headers`, `health-check`).
+
+**type** (required):
+- `base` - For main NGINX configuration templates
+- `augment` - For templates that extend base configurations
+
+**For Augment Templates Only**:
+- **allowed_in_contexts** - Specify which contexts this augment can be included in (e.g., `http`, `http/server`, `http/server/location`). Understanding where your augment fits within NGINX configuration structure is crucial for proper integration. See the [Template Authoring Guide]({{< ref "author-templates.md" >}}) for detailed information on contexts and how to design augments that integrate correctly with base templates.
+
+#### Import validation
+
+During import, the system will validate:
+- Archive structure (one .tmpl file, optional schema)
+- Template syntax
+- Schema validity (if provided)
+- Variable references match schema definitions
+
+{{< call-out "caution" >}}
+The `allowed_in_contexts` parameter is required when importing augment templates. The import process validates that this parameter is provided, but does not validate whether the contexts are valid for your template's NGINX directives. Context validation occurs during template submission. Ensure your specified contexts match where your NGINX directives can legally appear to avoid submission errors.
```

Comment:
```suggestion
The `allowed_in_contexts` parameter is required to import augment templates. The import process checks that this parameter is included, but it doesn’t confirm whether the contexts are valid for your template’s NGINX directives. Context validation happens during template submission. Ensure your specified contexts match where your NGINX directives are allowed to appear to avoid submission errors.
```